### PR TITLE
LOOP-4028 DIY Sync

### PR DESCRIPTION
**Pull request by ps2**
https://tidepool.atlassian.net/browse/LOOP-4028

* HealthKit queries started correctly when authorization not needed. No Tidepool Loop effects, since HK import is disabled.
* Support for Remote Bolus. No Tidepool Loop effects.
* Unify usage of fast-acting vs rapid-acting. No Tidepool Loop effects, as change only effects Insulin Selection screen.

### CODE REVIEW

File: LoopKit/CarbKit/CarbStore.swift [hunk 1]
```diff
@@ -252,7 +252,11 @@ public final class CarbStore: HealthKitSampleStore {
             cacheStore.fetchAnchor(key: CarbStore.healthKitQueryAnchorMetadataKey) { (anchor) in
                 self.queue.async {
                     self.queryAnchor = anchor
-            
+
+                    if !self.authorizationRequired {
```
Comment (rickpasetto) [inline LoopKit/CarbKit/CarbStore.swift:256]: @ps2 Maybe I'm missing something, but `!authorizationRequired` doesn't mean "authorization is not needed", but actually "authorization _is_ needed [but hasn't been granted yet]".  So, can you explain please why this (and all the others) are okay?  Thanks.

**Review by rickpasetto — COMMENTED** [review]
I have a question about the HK Query change before proceeding...thanks.

File: LoopKit/CarbKit/CarbStore.swift (hunk 1 above)
Comment (ps2) [reply LoopKit/CarbKit/CarbStore.swift:256]: Kinda. `!authorizationRequired` means "we've already asked the user for auth". In those cases where we've already asked, we can go ahead and start the query right here. If we haven't asked, we assume that eventually Loop will trigger the ask (`healthStore.requestAuthorization`), and then call `authorize()` on the stores, which will trigger the query creation.

Comment (rickpasetto) [reply LoopKit/CarbKit/CarbStore.swift:256]: Ah, ok, I think I see now.

**Review by rickpasetto — APPROVED** [review]
LGTM

**Review by darinkrauss — APPROVED** [review]
LGTM!

**Pull request merged**